src/TitleReceive/TitleReceiveContainer.js

```diff
@@ -160,9 +180,11 @@ function TitleReceiveContainer({ history, location, match, mutator, resources })
     [poLine],
   );
 
-  if (!(pieces && poLine && title)) return null;
+  if (!(pieces && poLine && title && pieceLocations && poLineLocationIds && poLineLocations)) return null;
+
   const initialValues = { receivedItems: pieces };
   const paneTitle = `${poLine.poLineNumber} - ${title.title}`;
+  const locations = [...new Set([...poLineLocations, ...pieceLocations])];
```

Comment:
please don't do things like this. It will always create a new array and pass it as a prop that will cause render of child components every time.